### ISSUE TITLE
Add an easy way to create empty layers of arbitrary dimensionality

### ISSUE THREAD

**Issue by jni** [napari/napari#1257]
## 🚀 Feature

A user on image.sc [found it hard](https://forum.image.sc/t/controlling-dimensionality-of-napari-point-shape-layers/37896) to create shapes layers of arbitrary dimensionality. Instead, `add_shapes` without data behaves differently depending on the state of the viewer, which is confusing. One may want, for example, to create a 2D shapes layer that broadcasts on a 3D movie, say, for calcium imaging data. It's entirely not obvious that you do this by passing an empty array of shape `(0, 0, ndim)`. A clearer way would be to add an `ndim` keyword argument that controlled dimensionality when no data is given.

**Comment by sofroniewn** [comment]
Indeed, we should improve this. Note this is basically a duplicate of #835 which still needs to be resolved

**Comment by jni** [comment]
Heh, always fun to disagree with my past self. 😂😅
